src/WebJobs.Script/Extensions/IDictionaryExtensions.cs

```diff
@@ -19,5 +20,27 @@ public static class IDictionaryExtensions
                 dictionary[pair.Key] = pair.Value;
             }
         }
+
+        public static IDictionary<string, TValue> AsCaseInsensitive<TValue>(this IDictionary<string, TValue> dictionary)
+        {
+            return new Dictionary<string, TValue>(dictionary, StringComparer.OrdinalIgnoreCase);
```

Comment:
Does this throw if there are dupe keys in the input dictionary? ie 'key' and 'KEY'.  Not sure if we'd prefer to throw or just overwrite one of the props in that case